_posts/blog/2016-07-07-reactive_mindset_burgers.md

```diff
@@ -0,0 +1,165 @@
+
+<center>
+<picture>
+	<a href="/images/blog/reactive_mindset_burgers/streams_of_data.png"><img src="/images/blog/reactive_mindset_burgers/streams_of_data.png" alt="streams of data" width="350"></a>
+	<figcaption>The buns, pieces of meat, slices of tomato and the burgers are streams of data</figcaption>
+</picture>
+</center>
+
+Donald Duck's customer, Mickey Mouse, he gets to consume the burger. So, Mickey Mouse is the one that acts upon the emitted items by the burger Observable. In ReactiveX, this is the ``Subscriber`` class.
+
+The actions that we performed on the stream of meat given by Dewey: filtering the meat that has gone bad and cooking the meat - so applying a function to every piece of meat - are called ``operators``.
+
+When developing using ReactiveX you have to switch from the pull based approach to a push based one. More precisely, Donald Duck does not pull buns from Huey. He does not request buns. Rather, Donald is just waiting for buns that are delivered to him by his nephew.
+
+<center>
+<picture>
+	<a href="/images/blog/reactive_mindset_burgers/pull_bun.png"><img src="/images/blog/reactive_mindset_burgers/pull_bun.png" alt="pull based development" width="350"></a>
+	<figcaption>Donald Duck doesn't request buns from Huey</figcaption>
+</picture>
+</center>
+
+We don't have an exact set of data that we can operate on, but rather we operate on each emission of a stream. This means that if we want to work only on meat that has not gone bad, we cannot do:
+{% highlight java %}
+for(Meat meat: meatList){
+    if(isMeatGoneBad(meat)){
+        meatList.remove(meat);
+    }
+}
+
+{% endhighlight %}
+because we don't have that ``meatList``. We have an Observable that emits meat pieces and from those, we are only interested in cooking the pieces that have not gone bad, those are the ones that we need to ``filter`` out.
+{% highlight java %}
+Observable.from(meatSource)
+          .filter(meat -> isMeatGoneBad(meat))
+{% endhighlight %}
+
+Similar, when we want to cook the meat, we cannot just apply a function to a set of data, because we don't have that set of data.
+{% highlight java %}
+for(Meat meat: meatList){
+    meat = cook(meat);
+}
+{% endhighlight %}
+
+But what we can do, is to apply a function to every emission of the filtered meat stream, by using the ``map`` operator.
+{% highlight java %}
+Observable.from(meatSource)
+          .filter(meat -> isMeatGoneBad(meat))
+          .map(meat -> cook(meat));
+{% endhighlight %}
+
+## Make a Reactive Burger
+
+Huey, Dewey and Louie create three ``Observable``s: bunObservable, meatObservable and the tomatoObservable. Donald will create the burgerObservable:
```

Comment:
i want to emphasise that the burgerObservable is created by zipping the other 3, that's why I want to mention all of them.